app/utils/database.py

```diff
@@ -379,5 +430,60 @@ def fetch_user_info(user_id):
     try:
         return User.get(User.uuid == user_id)
     except DoesNotExist:
-        logging.error("could not find user_id = %s", user_id)
+        return None
+
+def get_genebank(genebank_id, user_uuid=None):
+    """
+    Returns the information about the genebank given by `genebank_id` that is
+    accessible to the user identified by `user_uuid`.
+    """
+    if user_uuid is None:
+        return None
+    try:
+        genebank = Genebank.get(genebank_id).as_dict()
+        herd_query = Herd(database=DATABASE).select() \
+                                            .where(Herd.genebank == genebank_id)
+        genebank['herds'] = [h.as_dict() for h in herd_query.execute()]
+        return genebank
+    except DoesNotExist:
+        return None
+
+def get_genebanks(user_uuid=None):
+    """
+    Returns all genebanks that are accessible to the user identified by
+    `user_uuid`.
+    """
+    if user_uuid is None:
+        return None
+    genebanks_query = Genebank(database=DATABASE).select()
+    return [g.as_dict() for g in genebanks_query.execute()]
+
+def get_herd(herd_id, user_uuid=None):
+    """
+    Returns information on the herd given by `herd_id`, including a list of all
+    individuals belonging to that herd. The returned data is limited to the
+    access permission of the user identified by `user_uuid`.
+    """
+    if user_uuid is None:
+        return None
+    try:
+        herd = Herd.get(herd_id).as_dict()
+        individuals_query = Individual(database=DATABASE).select() \
+                                .where(Individual.herd == herd_id)
+        herd['individuals'] = [i.short_info() for i in individuals_query.execute()]
```

Comment:
I would just call `individuals_query` `query` (or even `q`) since its scope is just two lines :nerd_face: Then it's more apparent that the code is copied-and-pasted if the same name is used in `get_genebanks` above too.

Reply:
Haha - it's honestly part of an at times misguided attempt to use more descriptive variable names! 😁 I agree that `query` is a better name though!